src/containers/NFT/NFTHeader/actions.js

```diff
@@ -0,0 +1,37 @@
+import { getNFT } from '../../../rippled';
+import { analytics, ANALYTIC_TYPES, BAD_REQUEST, HASH_REGEX } from '../../shared/utils';
+import * as actionTypes from './actionTypes';
+
+export const loadNFTState = (tokenId, rippledSocket) => dispatch => {
+  if (!HASH_REGEX.test(tokenId)) {
+    dispatch({
+      type: actionTypes.NFT_STATE_LOAD_FAIL,
+      status: BAD_REQUEST,
+      error: '',
+    });
+    return Promise.resolve();
```

Comment:
why wouldn't this be `Promise.reject`? Not sure what the pattern is for redux here.

Reply:
the component picks up the state change of `status`, and renders error from there